### ISSUE TITLE
JDI crashes on preprocessing with invalid c++

### ISSUE THREAD

**Issue by RemoveRusky** [enigma-dev/enigma-dev#1186]
the following (invalid c++) will crash jdi. I noticed this when I forgot to terminate a /*

#ifdef JUST_DEFINE_IT_RUN
#error lolwut
#endif
class jdi : public isATurd {

jdi crashes with or without the JUST_DEFINE_IT_RUN macro

**Comment by JoshDreamland** [comment]
JDI could use some tests for invalid C++. In general, this isn't a huge problem for ENIGMA. I will see if this is still a problem in V2, which I plan to integrate into master in the coming weeks.

**Comment by RemoveRusky** [comment]
Yes, it's not a huge issue but a dev less familar with enigma's quirks might bang their head against wall when something like this happens so figured I'd detail it here.

**Comment by RobertBColton** [comment]
I decided to give this a test on the latest master since #1659 with the new JDi was merged. I copied the code fundies provided to the top of `GSmatrix.h` and it did not seem to crash JDI but did produce a lot of errors. Enough to freeze LGM's UI and require me to kill it with task manager. I tweaked it around a little bit and removed the inheritance but left the missing closing brace, restarted LGM, and tried compiling it again. It failed but that time did not freeze the UI and I can still continue hitting run again.

So it seems the original issue is gone, JDI is not crashing. However, LGM doesn't seem to handle JDI shitting a brick very well still as this freezing of the UI can occur randomly when editing the sources if you make mistakes. Considering that, I don't know if we want to keep this open. Do we consider JDI or LGM at fault now?

**Comment by fundies** [comment]
I think the hanging is related #1673. JDI seems to hangon mismatching brackets
